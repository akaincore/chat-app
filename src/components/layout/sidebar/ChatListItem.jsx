import React from 'react';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import ChatAvatar from '../../common/ChatAvatar';

const ChatListItem = props => {
  return (
    <ListItem
      button
      selected={props.selected}
      onClick={props.onClick}
    >
      <ChatAvatar
        title={props.title}
      />
      <ListItemText
        primary={props.title}
        secondary={props.hint}
      />
    </ListItem>
  );
};

export default ChatListItem;
