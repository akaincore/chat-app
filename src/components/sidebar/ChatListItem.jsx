import React from 'react';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import Avatar from '@material-ui/core/Avatar';
import initials from "../../utils/initials";

const ChatListItem = props => {
  return (
    <ListItem
      button
      selected={props.selected}
      onClick={props.onClick}
    >
      <Avatar>
        {initials(props.title)}
      </Avatar>
      <ListItemText
        primary={props.title}
        secondary={props.hint}
      />
    </ListItem>
  );
};

export default ChatListItem;
