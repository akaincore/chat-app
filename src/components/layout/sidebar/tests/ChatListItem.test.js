/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import ChatListItem from '../ChatListItem';

jest.mock('../../../common/ChatAvatar', () => () => 'ChatAvatar');

const mockProps = {
  chatId: '12345',
  selected: false,
  title: 'foo',
  createdAt: '2019-01-01T00:00:00.000Z',
};

describe('<ChatListItem />', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter>
        <ChatListItem {...mockProps} />
      </MemoryRouter>,
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <ChatListItem {...mockProps} />
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
