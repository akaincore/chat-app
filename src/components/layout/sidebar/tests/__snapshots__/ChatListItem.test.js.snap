// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatListItem /> renders correctly 1`] = `
<a
  className="MuiButtonBase-root-77 MuiListItem-root-65 MuiListItem-default-68 MuiListItem-gutters-73 MuiListItem-button-74"
  href="/chat/12345"
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="button"
  tabIndex="0"
>
  ChatAvatar
  <div
    className="MuiListItemText-root-80"
  >
    <span
      className="MuiTypography-root-86 MuiTypography-subheading-93 MuiListItemText-primary-83"
    >
      foo
    </span>
    <p
      className="MuiTypography-root-86 MuiTypography-body1-95 MuiTypography-colorTextSecondary-119 MuiListItemText-secondary-84"
    >
      11 days ago
    </p>
  </div>
  <span
    className="MuiTouchRipple-root-122"
  />
</a>
`;
